modules/images/webp-uploads/load.php

```diff
@@ -4,41 +4,414 @@
+ */
+function webp_uploads_valid_image_mime_types() {
+	$valid_formats = array(
+		'image/jpeg' => 'jpg',
+		'image/webp' => 'webp',
+	);
+
+	/**
+	 * An array representing all the valid mime types where multiple images would be created if
+	 * the image does not exist in that mime type.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array<string, string> $valid_formats array with the mime type as the key and extension as the value.
+	 * @return array<string, string> array with the mime type as the key and extension as the value.
+	 */
+	return (array) apply_filters( 'webp_uploads_images_with_multiple_mime_types', $valid_formats );
+}
+
+/**
+ * Callback used to return a callback used by `webp_uploads_ajax_wp_die_handler` filter.
+ *
+ * @since n.e.x.t
+ *
+ * @see   wp_die_ajax_handler
+ *
+ * @param callable $callback A callback executed after the images operation (restore, creation).
+ *
+ * @return callback A function that would be used as a callback.
+ */
+function webp_uploads_wp_die_ajax_handler( $callback ) {
+	if (
+		isset( $_REQUEST['action'], $_REQUEST['do'], $_REQUEST['postid'] )
+		&& 'image-editor' === $_REQUEST['action']
+		&& is_numeric( $_REQUEST['postid'] )
+	) {
+		$attachment_id = absint( $_REQUEST['postid'] );
+		switch ( $_REQUEST['do'] ) {
+			case 'restore':
+				webp_uploads_restore_image_from_backup( $attachment_id );
+				break;
+			case 'save':
+				webp_uploads_update_background_sizes( $attachment_id );
+				break;
+		}
+	}
+
+	return $callback;
+}
+
 /**
- * Filter the image editor default output format mapping.
+ * Update the keys on the backup sizes, by reordering the keys in the meta `_wp_attachment_backup_sizes`,
+ * files are removed only if the constant `IMAGE_EDIT_OVERWRITE` is defined to `true` and if the files
+ * were edited files, this pattern follows the same criteria used by WordPress core to handle edited
+ * images.
+ *
+ * @since n.e.x.t
  *
- * For uploaded JPEG images, map the default output format to WebP.
+ * @see   webp_uploads_ajax_wp_die_handler
  *
- * @since 1.0.0
+ * @param int $attachment_id ID of the attachment being modified.
+ */
+function webp_uploads_restore_image_from_backup( $attachment_id ) {
+	$metadata = wp_get_attachment_metadata( $attachment_id );
+	// This should take place only on the JPEG image.
+	$valid_mime_types = webp_uploads_valid_image_mime_types();
+
+	// Not a supported mime type to create the sources' property.
+	if ( ! array_key_exists( get_post_mime_type( $attachment_id ), $valid_mime_types ) ) {
+		return;
+	}
+
+	// If no size is present there's no need to generate any additional image as a backup.
+	if ( empty( $metadata['sizes'] ) ) {
+		// TODO: Maybe delete the meta key and files depending on `IMAGE_EDIT_OVERWRITE` ? if no size is present.
+		return;
+	}
+
+	// All subsizes are created out of the `file` property and not the original image.
+	$file = get_attached_file( $attachment_id, true );
```

Comment:
Why do you use get_attached_file here instead of the original image, is this to use the -scaled image?

Reply:
Correct any edits to an image are performed to the `scaled` version (if present) by WordPress so any subsize should be based out of the modified image and not the original image instead. 

The original image is only used to create the scaled version if not required it would be used as the attached file instead.